.circleci/config.yml

```diff
@@ -30,130 +30,133 @@ commands:
           key: node-cache-{{ checksum "package.json" }}-{{ checksum "yarn.lock" }}
           paths:
             - node_modules
-  run_rubocop:
-    description: 'Run rubocop'
+  webpacker_compile:
     steps:
       - restore_cache:
           keys:
-            - bundler-cache-{{ checksum ".ruby-version" }}-{{ checksum "Gemfile.lock" }}
+            - webpacker-cache-{{ .Branch }}
+            - webpacker-cache-main
+      - run:
+          name: 'Compile webpacker assets'
+          command: |
+            set -euxo pipefail
+            bundle config path .bundle
+            bundle exec rails webpacker:compile
+      - save_cache:
+          key: webpacker-cache-{{ .Branch }}
```

Comment:
In our meeting today @exbinary brought up this problem, where caches are write-once:

![Screen Shot 2021-05-08 at 11 08 21 AM](https://user-images.githubusercontent.com/3620291/117544118-d4f70300-afed-11eb-8944-08830995a659.png)

I dug into this and ran a few test runs on a separate branch.

It looks to me like the problem here is that webpack is putting everything in one unitary file (one file each for css & js). It doesn't split anything up and doesn't persist any useful intermediary steps to disk that we can then cache.

We can configure webpack differently to get something cacheable<sup>[1]</sup>. I'm not sure how long that would take, though. As a theoretical example, it should be possible to have one file that's all the packages we're pulling in and another that's only the code we've written that then pulls in the other<sup>[2]</sup>. I'm worried how hard it might be to get Rails to play nice with this, though<sup>[3]</sup>.

[1]: https://webpack.js.org/guides/code-splitting/#
[2]: https://dev.to/nejremeslnici/split-your-webpacker-bundles-to-speed-up-the-web-2
[3]: https://dev.to/rossta/these-rails-apps-are-overpacking-their-javascript-bundles-3gl5